### PR TITLE
Implement Matrix4Tween.lerp()

### PR DESCRIPTION
**Pull request by alanlgardner**
Just an implementation of what this already claims to do.

### CODE REVIEW

**Comment by Hixie** [comment]
This is fantastic.

Any chance you could add a test? Just verifying that this works roughly as we'd expect is sufficient. Something like the last test in <https://github.com/flutter/flutter/blob/master/packages/flutter/test/animation/tween_test.dart> would be fine.

**Comment by alanlgardner** [comment]
Will do.

**Comment by alanlgardner** [comment]
I noticed my dart formatter changed the formatting of the test file, what style guide do you use?

**Comment by alanlgardner** [comment]
Also, I attempted to do a slerp implementation so we would have constant speed rotation, and the results were always off from my expectations by a couple percent. After trying for a bit yesterday I just gave up and went back to direct linear interpolation and left a TODO. I have a CL if you want to take a look.

**Comment by Hixie** [comment]
Style guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

Prefer longer lines over line breaks after operators, but in general it's a judgement call. Optimize for readability.

I'll trust your judgement on what the best algorithm is here; it looks like what you have is better than what we had before, so it's an improvement either way!

It's not ideal that the test is in the animation/ directory but the code in widgets/ but don't worry about it, we have more work to clean up there already anyway.

**Comment by alanlgardner** [comment]
Sounds good. Let me know if there is anything else I should change.

File: packages/flutter/test/animation/tween_test.dart [hunk 1]
```diff
@@ -5,12 +5,14 @@
...
 import 'package:flutter/animation.dart';
 import 'package:flutter/widgets.dart';
+import 'package:vector_math/vector_math_64.dart';
 
 void main() {
   test('Can chain tweens', () {
     final Tween<double> tween = new Tween<double>(begin: 0.30, end: 0.50);
     expect(tween, hasOneLineDescription);
-    final Animatable<double> chain = tween.chain(new Tween<double>(begin: 0.50, end: 1.0));
+    final Animatable<double> chain =
```
Comment (Hixie) [inline packages/flutter/test/animation/tween_test.dart]: please don't wrap these

File: packages/flutter/test/animation/tween_test.dart [hunk 2]
```diff
@@ -31,7 +33,8 @@ void main() {
   });
 
   test('SizeTween', () {
-    final SizeTween tween = new SizeTween(begin: Size.zero, end: const Size(20.0, 30.0));
+    final SizeTween tween =
```
Comment (Hixie) [inline packages/flutter/test/animation/tween_test.dart]: same here

File: packages/flutter/test/animation/tween_test.dart [hunk 3]
```diff
@@ -49,4 +52,28 @@ void main() {
...
+
+  test('Matrix4Tween', () {
+    final Matrix4 a = new Matrix4.identity();
+    final Matrix4 b = new Matrix4.identity()
+      ..translate(6.0, -8.0, 0.0)
+      ..scale(0.5, 1.0, 5.0);
+    final Matrix4Tween tween = new Matrix4Tween(begin: a, end: b);
+    expect(tween.lerp(0.0), equals(a));
+    expect(tween.lerp(1.0), equals(b));
+    expect(
```
Comment (Hixie) [inline packages/flutter/test/animation/tween_test.dart]: flutter style is that if you have a newline after the `(`, you should have one before the matching `)`, and the contents should be indented by 2.

File: packages/flutter/test/animation/tween_test.dart [hunk 4]
```diff
@@ -49,4 +52,28 @@ void main() {
...
+        tween.lerp(0.5),
+        equals(new Matrix4.identity()
+          ..translate(3.0, -4.0, 0.0)
+          ..scale(0.75, 1.0, 3.0)));
+    final Matrix4 c = new Matrix4.identity()..rotateZ(1.0);
+    final Matrix4Tween rotationTween = new Matrix4Tween(begin: a, end: c);
+    expect(rotationTween.lerp(0.0), equals(a));
+    expect(rotationTween.lerp(1.0), equals(c));
+    expect(
```
Comment (Hixie) [inline packages/flutter/test/animation/tween_test.dart]: ditto

**Comment by Hixie** [comment]
Just some stylistics nits.

**Comment by alanlgardner** [comment]
How do we get this check to re-run when the build is not broken anymore?

**Comment by Hixie** [comment]
It does so automatically.

**Comment by Hixie** [comment]
Thanks for the contribution! I will land your patch!

**Comment by alanlgardner** [comment]
Awesome, thanks for the help!

**Pull request merged**